src/APK/Podman.hs

```diff
@@ -0,0 +1,186 @@
+  , hostCachePath              :: Path.AbsDir
+  , hostHomePath               :: Path.AbsDir
+  , hostPackagesPath           :: Path.AbsDir
+  , hostRepositoryPackagesPath :: Path.AbsDir
+  , hostRepositoryPath         :: Path.AbsDir
+  , hostWorkingPath            :: Path.AbsDir
+  }
+
+defaultConfig :: Config
+defaultConfig = Config
+  { arguments      = []
+  , command        = "exit"
+  , repositoryPath = Path.relDir "repository"
+  , workPath       = containerHomePath
+  }
+
+getHostPaths :: IO HostPaths
+getHostPaths
+  =   mkHostPaths
+  <$> Path.getCurrentDirectory
+  <*> Path.getHomeDirectory
+  where
+    mkHostPaths :: Path.AbsDir -> Path.AbsDir -> HostPaths
+    mkHostPaths hostWorkingPath hostHomePath = HostPaths
+      { hostAbuildPath             = hostHomePath </> Path.relDir ".abuild"
+      , hostCachePath              = hostWorkingPath </> Path.relDir "cache"
+      , hostPackagesPath           = hostWorkingPath </> Path.relDir "packages"
+      , hostRepositoryPackagesPath = hostRepositoryPath </> Path.relDir "packages"
+      , ..
+      }
+      where
+        hostRepositoryPath :: Path.AbsDir
+        hostRepositoryPath = hostWorkingPath </> Path.relDir "repository"
+
+containerAbuildPath :: Path.AbsDir
+containerAbuildPath = containerHomePath </> Path.relDir ".abuild"
+
+containerHomePath :: Path.AbsDir
+containerHomePath = Path.absDir "/opt/build"
+
+containerCachePath :: Path.AbsDir
+containerCachePath = Path.absDir "/var/cache"
+
+containerRepositoryPath :: Path.AbsDir
+containerRepositoryPath = containerHomePath </> Path.relDir "repository"
+
+containerPackagesPath :: Path.AbsDir
+containerPackagesPath = containerHomePath </> Path.relDir "packages"
+
+run :: String -> [String] -> IO ()
+run command arguments = runConfig defaultConfig{arguments = arguments, command = command}
+
+index :: IO ()
+index =
+  runConfig defaultConfig
+    { arguments = ["-F", "index"]
+    , command   = "abuild"
+    , workPath  = containerPackagesPath </> Path.relDir "stack"
+    }
+
+update :: IO ()
+update = run "apk" ["update"]
+
+build :: PackageName -> IO ()
+build packageName =
+  runConfig defaultConfig
+    { arguments = ["-r", "-F"]
+    , command   = "abuild"
+    , workPath  = containerPackagesPath </> Path.relDir (convertText packageName)
+    }
+
+runConfig :: Config -> IO ()
+runConfig config = runContainer config =<< getHostPaths
+
+runContainer :: Config -> HostPaths -> IO ()
+runContainer Config{..} HostPaths{..} = do
+  buildImage
+
+  Path.createDirectoryIfMissing False hostRepositoryPath
+  Path.createDirectoryIfMissing False hostPackagesPath
+  Path.createDirectoryIfMissing False hostAbuildPath
+  Path.createDirectoryIfMissing False hostCachePath
+  Path.createDirectoryIfMissing False $ hostCachePath </> Path.relDir "apk"
+  Path.createDirectoryIfMissing False $ hostCachePath </> Path.relDir "distfiles"
+  Path.createDirectoryIfMissing False $ hostCachePath </> Path.relDir "misc"
```

Comment:
Could a utility method be extracted that removes the repeating `Path.createDirectoryIfMissing False`?